### ISSUE TITLE
why `model.train()` before DDP wrapper?

### ISSUE THREAD

**Issue by toyot-li** [open-mmlab/OpenLidarPerceptron#1118]
@sshaoshuai Thanks for the awesome toolbox. But I'm not quite clear about why [this line](https://github.com/open-mmlab/OpenPCDet/blob/master/tools/train.py#L154) here occurs before the DDP model wrapper, even with the simple comment. Could you explain more about this? Thanks.
For comparison, in some well-developed 2D detection codebases, DDP training for detectors with frozen parameters seems not to require such a workaround.

**Comment by sshaoshuai** [comment]
I forget the reason for this line. But you can try to move this code behind the DDP and see that whether you can fix some params. 
I just remember that it may fail to fix some params if we do not set the model to training mode before feeding to DDP.

**Comment by sshaoshuai** [comment]
I guess the reason may be related to here: https://github.com/pytorch/pytorch/blob/master/torch/nn/parallel/distributed.py#L816-L818, 
that is, if you doesn't set the params to training mode, it will not be wrapped by DDP.

**Comment by sshaoshuai** [comment]
That's a good question. I think you can try to move model.train() behind the DDP and see what happens, which may provide some hints about this problem. 
I did have a problem fixing some parameters if I write model.train() after DDP, but I am not sure whether this issue still exists since it has been a very long time since my last related attempt.
